### DISCUSSION TITLE
In what order are `LanguageServerFactory` and `LanguageServerEnablementSupport`'s methods called?

### DISCUSSION THREAD

**Discussion by InSyncWithFoo** [redhat-developer/lsp4ij#335] [General]
I'm currently overriding `isEnabled()`, `setEnabled()`, `createConnectionProvider()` and `createLanguageClient()`. The third of which must return an instance of `StreamConnectionProvider` without fail. However, this can only success depending on a specific user setting.

Is `isEnabled()` guaranteed to always be called before `createConnectionProvider()`? When are the other methods called? 

It would be useful to have more information about these in doc comments. They are currently stating only what should be obvious from the method names alone:

> Returns true if the language server is enabled for the given project and false otherwise.

**Comment by angelozerr** [comment]
IsEnabled is used while collecting associated language servers for a given file.

This step is done each time that an lsp feature like completion is processed. 

This step is done before starting the language server, in other words before creating a connection or a client.

**Reply by angelozerr** [reply]
@InSyncWithFoo I apologize but my PR with isEnabled was totally bugged, I created a PR https://github.com/redhat-developer/lsp4ij/pull/349 to fix it.

**Comment by angelozerr** [comment]
I close this discussion since isEnabled work correctly now.
